src/main/java/io/vertx/ext/mongo/impl/Utils.java

```diff
@@ -57,12 +57,19 @@ static MongoClientBulkWriteResult toMongoClientBulkWriteResult(BulkWriteResult b
       return upsertValue;
     }).collect(Collectors.toList());
 
+    List<JsonObject> insertResult = bulkWriteResult.getInserts().stream().map(insert -> {
+      JsonObject insertValue = convertInsertId(insert.getId());
+      insertValue.put(MongoClientBulkWriteResult.INDEX, insert.getIndex());
+      return insertValue;
+    }).collect(Collectors.toList());
+
```

Comment:
Can you extract this code to a private method and reuse it above?

Reply:
It is not possible to extract complete logic to method as getUpserts and getInserts are returning different object types BulkWriteUpsert/BulkWriteInsert but part of the  logic can be extracted.